apps/server/test/lexical/server/code_intelligence/entity_test.exs

```diff
@@ -299,30 +304,247 @@ defmodule Lexical.Server.CodeIntelligence.DefinitionTest do
     test "find the definition when calling a erlang module", %{project: project} do
       subject_module = ~q[
         :erlang.binary_to_atom|("1")
-      ]
+        ]
       {:ok, uri, definition_line} = definition(project, subject_module)
 
       assert uri =~ "/src/erlang.erl"
       assert definition_line =~ ~S[«binary_to_atom»(Binary)]
     end
   end
 
-  defp caller_position(subject_module) do
-    {line, character} = cursor_position(subject_module)
-    Position.new(line, character)
+  describe "resolve/2" do
+    test "syntax error", %{project: project} do
+      code = ~q[
+        Some.Modul|e.
+        ]
+
+      assert {:error, _} = resolve(project, code)
+    end
+
+    test "returns an error if the cursor is not over an entity", %{project: project} do
+      code = ~q[
+        Beyond.The.End|
+        ]
+
+      assert {:error, :not_found} = resolve(project, code)
+
+      code = ~q[
+        | Beyond.The.End
+        ]
+
+      assert {:error, :not_found} = resolve(project, code)
+    end
+
+    test "returns module entities at and before the cursor", %{project: project} do
+      code = ~q[
+        At.The.En|d
+        ]
+
+      assert {:ok, {:module, At.The.End}} = resolve(project, code)
+
+      code = ~q[
+        Beginning.Of.The|.End
+        ]
+
+      assert {:ok, {:module, Beginning.Of.The}} = resolve(project, code)
+
+      code = ~q[
```

Comment:
I don't see the difference between this case and `At.The.En|d`.

I even feel that repeating such things is unnecessary. Tests require extreme focus and should have clear differentiation.